src/AutoRest.CSharp/Common/Input/Source/ProtocolCompilationInput.cs

```diff
@@ -93,6 +94,9 @@ private protected List<IMethodSymbol> EnsureMethodSet()
         }
 
         private bool IsClient(INamedTypeSymbol type) => type.Name.EndsWith("Client");
-        private bool IsProtocolMethod(IMethodSymbol method) => method.Parameters.Length > 0 && method.Parameters.Last().Name == "context";
+        private bool IsProtocolMethod(IMethodSymbol method) =>
+            method.Parameters.Length > 0
+            && ((method.Parameters.Last().Type as INamedTypeSymbol)!.IsSameType(KnownParameters.RequestContext.Type)
+            || (method.Parameters.Last().Type as INamedTypeSymbol)!.IsSameType(KnownParameters.RequestContextRequired.Type));
```

Comment:
shall we have a local variable to get the last parameter type?

Reply:
What is the value? `method.Parameters.Last()` is an `O(1)` operation.